task/golangci-lint/0.1/golangci-lint.yaml

```diff
@@ -0,0 +1,43 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: golangci-lint
+  labels:
+    app.kubernetes.io/version: "0.1"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.36.0"
+    tekton.dev/categories: Code Quality
+    tekton.dev/tags: linter
+    tekton.dev/displayName: "golangci-lint"
+    tekton.dev/platforms: "linux/amd64"
+spec:
+  description: >-
+    The golangci-lint Task runs a Golang linter.
+    The task is tailored to be triggered by Prow as a ProwJob.
+    It uses a golangci-lint tool.
+    The GolangCI tool config is stored in the `.golangci.yaml` file in a repository hosting tested code.
+    The task does not accept additional parameters to configure the linter. By default, it runs linter against all modules found under task working directory.
+  workspaces:
+    - name: repo
+      description: "The workspace where sources to lint are stored. Usually, it should be a workspace shared with other tasks. A default mount path is aligned with a ProwJob running under a Kubernetes agent."
+      mountPath: /home/prow/go
+  params:
+    - name: REPO_OWNER
+      description: "GitHub org that triggers the job. A variable set by Prow. Prow does not set this variable if the ProwJob type is periodic. In that case, an empty string default value is used."
+      type: string
+      default: ""
+    - name: REPO_NAME
+      description: "GitHub repo that triggers the job. A variable set by Prow. Prow does not set this variable if the ProwJob type is periodic. In that case, an empty string default value is used."
+      type: string
+      default: ""
+  steps:
+    - name: lint-go-code
+      image: eu.gcr.io/kyma-project/test-infra/golangci-lint:v20230221-85b2d35f
+      workingDir: /home/prow/go/src/github.com/$(params.REPO_OWNER)/$(params.REPO_NAME)
+      timeout: 30m
```

Comment:
Timeout should be configurable by parameters if possible. I would really insist on that because mileage may vary for different people.

Reply:
timeout can't be set by params. A task definition with a param reference as a value for timeout failed validation with following messages.
```shell
admission webhook "webhook.pipeline.tekton.dev" denied the request: mutation failed: cannot decode incoming new object: time: invalid duration "$(params.task-timeout)"
```
I've removed setting this value in a Task in favor of timeout defined in config-defaults config map. I will increase a default value to 120 minutes to match a timeout set on prowjobs running on kubernetes agent.